pkg/api/k8s_versions.go

```diff
@@ -165,17 +165,21 @@ var k8sComponentVersions = map[string]map[string]string{
 	},
 }
 
-// K8sComponentsByVersionMap represents Docker images used for Kubernetes components based on Kubernetes versions (major.minor.patch)
-var K8sComponentsByVersionMap map[string]map[string]string
+type getK8sVersionComponents func(string, map[string]string) map[string]string
 
-func init() {
-	K8sComponentsByVersionMap = getKubeConfigs()
-}
-
-func getKubeConfigs() map[string]map[string]string {
+func GetK8sComponentsByVersionMap(k *KubernetesConfig) map[string]map[string]string {
```

Comment:
... and we refactor the existing exported static dictionary into the response from an exported `GetK8sComponentsByVersionMap`function, which will return the appropriate components dictionaries based on the value of `KubernetesImageBaseType`.